core/src/main/java/org/infinispan/LockedStream.java

```diff
@@ -84,6 +88,55 @@ default void forEach(SerializableBiConsumer<Cache<K, V>, ? super CacheEntry<K, V
    }
 
    /**
+    * Performs a function for each element of this stream on the primary owner of the given key returning
+    * a value. The returned value from the function will be sent back to the user mapped to the key that generated
+    * it, with all of these stored in a map. This method will return only after all entries have been processed.
+    * <p>
+    * This method is currently marked as {@link Experimental} since this method returns a Map and requires blocking.
+    * This operation could take a deal of time and as such should be done using an asynchronous API. Most likely
+    * this return type will be changed to use some sort of asynchronous return value. This method is here until
+    * this can be implemented.
+    * <p>
+    * This method is performed while holding exclusive lock over the given entry and will be released
+    * only after the function has completed. If the entry is directly modified via the
+    * {@link java.util.Map.Entry#setValue(Object)} method this will be the same as if
+    * {@link java.util.Map#put(Object, Object)} was invoked.
+    * <p>
+    * If using a pessimistic transaction this lock is not held using a transaction and thus the user can start a
+    * transaction in this consumer which also must be completed before returning. A transaction can be started in
+    * the consumer and if done it will share the same lock used to obtain the key.
+    * <p>
+    * Remember that if you are using an explicit transaction or an async method that these must be completed before
+    * the consumer returns to guarantee that they are operating within the scope of the lock for the given key. Failure
+    * to do so will lead into possible inconsistency as they will be performing operations without the proper locking.
+    * <p>
+    * Some methods on the provided cache may not work as expected. These include
+    * {@link AdvancedCache#putForExternalRead(Object, Object)}, {@link AdvancedCache#lock(Object[])},
+    * {@link AdvancedCache#lock(Collection)}, and {@link AdvancedCache#removeGroup(String)}.
+    * If these methods are used inside of the Consumer on the cache it will throw a {@link IllegalStateException}.
+    * This is due to possible interactions with transactions while using these commands.
+    * @param biFunction the biFunction to run for each entry under their lock
+    * @param <R> the return type
+    * @return a map with each key mapped to the value returned from the bi function
+    */
+   @Experimental
+   <R> Map<K, R> invokeAll(BiFunction<Cache<K, V>, ? super CacheEntry<K, V>, R> biFunction);
```

Comment:
`invokeAll` sounds to me like it should receive a collection of functions and invoke all of them.

How about `forEachMap`? It would make it clearer that the function can have side effects, just like `forEach`.

Reply:
We were thinking it would be more closely named to what `JCache` provides https://static.javadoc.io/javax.cache/cache-api/1.0.0/javax/cache/Cache.html#invokeAll(java.util.Set, javax.cache.processor.EntryProcessor, java.lang.Object...)
I am personally not tied to any name.

Reply:
Let's keep `invokeAll`, using the same name as JCache is a good enough reason to me.

Reply:
Yeah and it is marked as Experimental, so we can always change it later :)